src/honeysql_postgres/helpers.cljc

```diff
@@ -38,6 +39,17 @@
 (defhelper with-columns [m args]
   (assoc m :with-columns args))
 
+(defhelper constraint [m args]
```

Comment:
A couple of suggestions here

1. Use `defhelper` only to update the query map with the constraint clauses and not to construct the SQL function call. We want to utilise the format-clause multimethod to construct the function call so that it works with [honeysql's priority-based ordering rule system](https://github.com/seancorfield/honeysql/blob/develop/src/honeysql/format.cljc#L262). Also, add a [priority](https://github.com/nilenso/honeysql-postgres/blob/master/src/honeysql_postgres/format.cljc#L7) to the constraint clause, which I think would be a number slightly larger than the priority assigned for `with-columns`. I just noticed that `with-columns` doesn't have a priority assigned to it either,  but it will need one now. Could you please add one for it as well? 
2. Pluralise this function's name and have it support multiple constraints so that going forward we're able to use this function for defining multiple constraints.  ```
(constraints [:unique [:product_no :product_sku]]
                    [:primary-keys [:name]])
```
3. Make constraints a top-level key in the query map instead of `assoc`ing it with the columns. It otherwise ends up treating the constraint as a column.